packages/flutter/lib/src/material/carousel.dart

```diff
@@ -480,6 +576,511 @@ class _RenderSliverFixedExtentCarousel extends RenderSliverFixedExtentBoxAdaptor
+        // missing item. Let parent re-layout everything.
+        geometry = SliverGeometry(scrollOffsetCorrection: indexToLayoutOffset(deprecatedExtraItemExtent, index));
+        return;
+      }
+      final SliverMultiBoxAdaptorParentData childParentData = child.parentData! as SliverMultiBoxAdaptorParentData;
+      childParentData.layoutOffset = indexToLayoutOffset(deprecatedExtraItemExtent, index);
+      assert(childParentData.index == index);
+      trailingChildWithLayout ??= child;
+    }
+
+    if (trailingChildWithLayout == null) {
+      firstChild!.layout(_getChildConstraints(indexOf(firstChild!)));
+      final SliverMultiBoxAdaptorParentData childParentData = firstChild!.parentData! as SliverMultiBoxAdaptorParentData;
+      childParentData.layoutOffset = indexToLayoutOffset(deprecatedExtraItemExtent, firstIndex);
+      trailingChildWithLayout = firstChild;
+    }
+
+    // From the last item to the firstly encountered max item
+    double extraLayoutOffset = 0;
+    if (allowFullyExpand) {
+      for (int i = weights.length - 1; i >= 0; i--) {
+        if (weights.elementAt(i) == weights.max) {
+          break;
+        }
+        extraLayoutOffset += weights.elementAt(i) * extentUnit;
+      }
+    }
+
+    double estimatedMaxScrollOffset = double.infinity;
+    // Layout visible items after the first visible item.
+    for (int index = indexOf(trailingChildWithLayout!) + 1; targetLastIndex == null || index <= targetLastIndex; ++index) {
+      RenderBox? child = childAfter(trailingChildWithLayout!);
+      if (child == null || indexOf(child) != index) {
+        child = insertAndLayoutChild(_getChildConstraints(index), after: trailingChildWithLayout);
+        if (child == null) {
+          // We have run out of children.
+          estimatedMaxScrollOffset = indexToLayoutOffset(deprecatedExtraItemExtent, index) + extraLayoutOffset;
+          break;
+        }
+      } else {
+        child.layout(_getChildConstraints(index));
+      }
+      trailingChildWithLayout = child;
+      final SliverMultiBoxAdaptorParentData childParentData = child.parentData! as SliverMultiBoxAdaptorParentData;
+      assert(childParentData.index == index);
+      childParentData.layoutOffset = indexToLayoutOffset(deprecatedExtraItemExtent, childParentData.index!);
+    }
+
+    final int lastIndex = indexOf(lastChild!);
+    final double leadingScrollOffset = indexToLayoutOffset(deprecatedExtraItemExtent, firstIndex);
+    double trailingScrollOffset;
+
+    if (lastIndex + 1 == childManager.childCount) {
+      trailingScrollOffset = indexToLayoutOffset(deprecatedExtraItemExtent, lastIndex);
+
+      trailingScrollOffset += math.max(weights.last * extentUnit, _buildItemExtent(lastIndex, currentLayoutDimensions));
+      trailingScrollOffset += extraLayoutOffset;
+    } else {
+      trailingScrollOffset = indexToLayoutOffset(deprecatedExtraItemExtent, lastIndex + 1);
+    }
+
+    assert(debugAssertChildListIsNonEmptyAndContiguous());
```

Comment:
Should the debug assert not be named `debugAssertChildListIsNonEmptyAndContinuous` ?